ballsdex/packages/balls/cog.py

```diff
@@ -738,3 +738,122 @@ async def duplicate(
 
         paginator = Pages(source, interaction=interaction)
         await paginator.start(ephemeral=True)
+
+    @app_commands.command()
+    @app_commands.checks.cooldown(1, 60, key=lambda i: i.user.id)
+    async def compare(
+        self,
+        interaction: discord.Interaction["BallsDexBot"],
+        user: discord.User,
+        special: SpecialEnabledTransform | None = None,
+    ):
+        """
+        Compare your countryballs with another user.
+
+        Parameters
+        ----------
+        user: discord.User
+            The user you want to compare with
+        special: Special
+            Filter the results of autocompletion to a special event. Ignored afterwards.
+        """
+        await interaction.response.defer(thinking=True)
+        if interaction.user == user:
+            await interaction.followup.send("You cannot compare with yourself.", ephemeral=True)
+            return
+
+        try:
+            player = await Player.get(discord_id=user.id)
+        except DoesNotExist:
+            await interaction.followup.send(
+                f"{user.display_name} doesn't have any {settings.plural_collectible_name} yet."
+            )
+            return
+
+        if await inventory_privacy(self.bot, interaction, player, user) is False:
+            return
+
+        bot_countryballs = {x: y.emoji_id for x, y in balls.items() if y.enabled}
+        if special:
+            bot_countryballs = {
+                x: y.emoji_id
+                for x, y in balls.items()
+                if y.enabled and (special.end_date is None or y.created_at < special.end_date)
+            }
+
+        player1, _ = await Player.get_or_create(discord_id=interaction.user.id)
+        player2, _ = await Player.get_or_create(discord_id=user.id)
+
+        blocked = await player.is_blocked(player1)
+        if blocked and not is_staff(interaction):
+            await interaction.followup.send(
+                "You cannot compare with a user that has you blocked.", ephemeral=True
+            )
+            return
+
+        blocked = await player.is_blocked(player2)
+        if blocked and not is_staff(interaction):
+            await interaction.followup.send(
+                "You cannot compare with a user that has you blocked.", ephemeral=True
+            )
+            return
+
+        user1_balls = cast(
+            list[int],
+            await player1.balls.filter(special=special).values_list("ball_id", flat=True),
+        )
+        user2_balls = cast(
+            list[int],
+            await player2.balls.filter(special=special).values_list("ball_id", flat=True),
+        )
+        both = set(user1_balls) & set(user2_balls)
+        user1_only = set(user1_balls) - set(user2_balls)
+        user2_only = set(user2_balls) - set(user1_balls)
+        neither = set(balls.keys()) - both - user1_only - user2_only
```

Comment:
Needs to be based on our own dictionnary, otherwise extra unwanted keys appear
```suggestion
        neither = set(bot_countryballs.keys()) - both - user1_only - user2_only
```